homeassistant/components/efergy/sensor.py

```diff
@@ -1,90 +1,106 @@
     ),
-    CONF_AMOUNT: SensorEntityDescription(
-        key=CONF_AMOUNT,
-        name="Energy Consumed",
+    SensorEntityDescription(
+        key="energy_day",
+        name="Daily Consumption",
         device_class=DEVICE_CLASS_ENERGY,
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        entity_registry_enabled_default=False,
     ),
-    CONF_BUDGET: SensorEntityDescription(
-        key=CONF_BUDGET,
+    SensorEntityDescription(
+        key="energy_week",
+        name="Weekly Consumption",
+        device_class=DEVICE_CLASS_ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="energy_month",
+        name="Monthly Consumption",
+        device_class=DEVICE_CLASS_ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    SensorEntityDescription(
+        key="energy_year",
+        name="Yearly Consumption",
+        device_class=DEVICE_CLASS_ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="budget",
         name="Energy Budget",
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="cost_day",
+        name="Daily Energy Cost",
+        device_class=DEVICE_CLASS_MONETARY,
+        entity_registry_enabled_default=False,
     ),
-    CONF_COST: SensorEntityDescription(
-        key=CONF_COST,
-        name="Energy Cost",
+    SensorEntityDescription(
+        key="cost_week",
+        name="Weekly Energy Cost",
         device_class=DEVICE_CLASS_MONETARY,
+        entity_registry_enabled_default=False,
     ),
-    CONF_CURRENT_VALUES: SensorEntityDescription(
+    SensorEntityDescription(
+        key="cost_month",
+        name="Monthly Energy Cost",
+        device_class=DEVICE_CLASS_MONETARY,
+    ),
+    SensorEntityDescription(
+        key="cost_year",
+        name="Yearly Energy Cost",
+        device_class=DEVICE_CLASS_MONETARY,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
         key=CONF_CURRENT_VALUES,
-        name="Per-Device Usage",
+        name="Power Usage",
         device_class=DEVICE_CLASS_POWER,
         native_unit_of_measurement=POWER_WATT,
     ),
-}
-
-TYPES_SCHEMA = vol.In(SENSOR_TYPES)
-
-SENSORS_SCHEMA = vol.Schema(
-    {
-        vol.Required(CONF_TYPE): TYPES_SCHEMA,
-        vol.Optional(CONF_CURRENCY, default=""): cv.string,
-        vol.Optional(CONF_PERIOD, default=DEFAULT_PERIOD): cv.string,
-    }
 )
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_APPTOKEN): cv.string,
-        vol.Optional(CONF_UTC_OFFSET, default=DEFAULT_UTC_OFFSET): cv.string,
-        vol.Required(CONF_MONITORED_VARIABLES): [SENSORS_SCHEMA],
-    }
+# Deprecated in Home Assistant 2021.11
+PLATFORM_SCHEMA = cv.deprecated(
```

Comment:
We can't use `cv.deprecated` like this. It requires a key. For platforms just log a warning during legacy platform setup.